### PR TITLE
[5.1] Fixed aggregate queries

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1749,21 +1747,24 @@ public function aggregate($function, $columns = ['*'])
...
 
     /**
-     * Format the return value of an aggregate function.
+     * Execute and format an aggregate function on the database.
      *
-     * @param  array  $results
-     * @return float|int
+     * @param  string  $function
+     * @param  array   $columns
+     * @return mixed
```
Comment (JoostK) [inline src/Illuminate/Database/Query/Builder.php]: This may now become `int|float`

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: Ah, yeh, oops, Meant to put that :)

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -1749,21 +1747,24 @@ public function aggregate($function, $columns = ['*'])
...
      */
-    protected function formatAggregate($results)
+    public function formatedAggregate($function, $columns = ['*'])
```
Comment (JoostK) [inline src/Illuminate/Database/Query/Builder.php]: Probably make that `formattedAggregate` (typo fix) or `numberAggregate` for more descriptive name.

**Comment by taylorotwell** [comment]
Can you describe whats going on here?

**Comment by JoostK** [comment]
I like that you try to do something with my earlier comments in #14781 but really doubt if we should go this way. Has there been any practical issues with `aggregate` how it always behaved?

**Comment by GrahamCampbell** [comment]
Yeh, it'll break if you have strict types turned on and you were expecting a float instead of a string without these changes.

**Comment by JoostK** [comment]
Ah, in a PHP 7 context that makes sense. With `formattedAggregate` having been changed to `numericalAggregate` this is then okay with me.

**Comment by segadora** [comment]
I might have posted a comment in the wrong place.

I have issues since this change with the min and max method. I am trying to get the maximum value if a date time field like `2016-08-24 00:00:00`, but the max() method only returns 2016. Is this the expected behavior and if so, how to get the min and max values of a date time field?

**Comment by GrahamCampbell** [comment]
Basically reverted this.

**Pull request merged**